### PR TITLE
Job Training Services Module

### PR DESCRIPTION
**Pull request by scfi0**

### CODE REVIEW

File: ServiceProviderBot/appsettings.Development.json [hunk 1]
```diff
@@ -16,5 +16,5 @@
   "ConnectionStrings": {
     "DbModel": "data source=(LocalDb)\\MSSQLLocalDB;initial catalog=OrganizationModel;integrated security=True;MultipleActiveResultSets=True;App=EntityFramework"
   },
-  "MapsKey": "<create personal key in azure. Will have one to share when Azure grant comes through>"
```
Comment (jcyriac2) [inline ServiceProviderBot/appsettings.Development.json]: We updated this with a key so that we can get map location services during development.

File: EntityModel/Migrations/20190228215829_InitialCreate.Designer.cs [hunk 2]
```diff
@@ -1,94 +0,0 @@
-﻿// <auto-generated />
```
Comment (JBrownie77) [inline EntityModel/Migrations/20190228215829_InitialCreate.Designer.cs]: You'll want to not delete these files. This is the migration from before the hackathon started. You can just revert the changes to these 3

File: ServiceProviderBot/Bot/Dialogs/UpdateOrganization/JobTraining/UpdateJobTrainingDialog.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        public UpdateJobTrainingDialog(StateAccessors state, DialogSet dialogs, DbInterface database, IConfiguration configuration)
+            : base(state, dialogs, database, configuration) { }
+
+        public override WaterfallDialog GetWaterfallDialog()
+        {
+            return new WaterfallDialog(Name, new WaterfallStep[]
+            {
+                async (stepContext, cancellationToken) =>
+                {
+                    var organization = await database.GetOrganization(stepContext.Context);
```
Comment (JBrownie77) [inline ServiceProviderBot/Bot/Dialogs/UpdateOrganization/JobTraining/UpdateJobTrainingDialog.cs]: Looks like organization isn't used here

File: ServiceProviderBot/ServiceProviderBot.csproj [hunk 4]
```diff
@@ -18,7 +18,6 @@
 
   <ItemGroup>
     <Folder Include="wwwroot\" />
-    <Folder Include="Bot\Dialogs\UpdateOrganization\" />
```
Comment (jcyriac2) [inline ServiceProviderBot/ServiceProviderBot.csproj]: Should this be removed?

File: ServiceProviderBot/Bot/Dialogs/UpdateOrganization/UpdateOrganizationDialog.cs [hunk 5]
```diff
@@ -39,6 +39,16 @@ public override WaterfallDialog GetWaterfallDialog()
                     return await BeginDialogAsync(stepContext, UpdateCapacityDialog.Name, null, cancellationToken);
                 },
                 async (stepContext, cancellationToken) =>
+                {
+                    var organization = await database.GetOrganization(stepContext.Context);
+                    if(organization.HasJobTrainingServices)
```
Comment (JBrownie77) [inline ServiceProviderBot/Bot/Dialogs/UpdateOrganization/UpdateOrganizationDialog.cs]: James added a NeedsUpdate method in his change that checks if the individual services need to update. If you pull the latest from develop into your branch (git pull --rebase origin develop) then you can add you check into that

File: EntityModel/Migrations/20190228215829_InitialCreate.Designer.cs (hunk 2 above)
Comment (jcyriac2) [reply EntityModel/Migrations/20190228215829_InitialCreate.Designer.cs]: Sorry @scfi0 . It was supposed to be `git checkout origin/develop -- path/to/file`. My bad 😅 :
